repository-structure.md

```diff
@@ -0,0 +1,72 @@
+# Common Repository Structure
+
+Hyperledger projects are required to maintain a standard set of files in each repository. This
+document describes the required and recommended files.
+
+## Required Files with Specified Content
+
+Repositories MUST have these files with the specific content in the linked files, or a file with a
+link to the specified content with minimal exposition. These files MUST be at the root of the
+repository.
+
+-   [`LICENSE`](https://www.apache.org/licenses/LICENSE-2.0.txt) - https://www.apache.org/licenses/LICENSE-2.0.txt
+-   [`CODE_OF_CONDUCT.md`](https://wiki.hyperledger.org/community/hyperledger-project-code-of-conduct) - https://wiki.hyperledger.org/community/hyperledger-project-code-of-conduct
+-   [`SECURITY.md`](https://wiki.hyperledger.org/display/SEC/Defect+Response) - https://wiki.hyperledger.org/display/SEC/Defect+Response
+
+## Required Files with Variable Content
+
+Repositories MUST have these files. Named files MUST be at the root of the repository, and may have
+format suffixes such as `.md`, `.rst`, or `.txt`.
+
+-   `README` \
+    A description of the project and contain information or links to information such as
+    -   A reference to the Apache license (required).
+    -   The current and important past releases
+    -   Documentation for developers and users
+-   `MAINTAINERS` \
+    A list of all current maintainers with contact info. [A separate document](MAINTAINERS-guidelines.md)
+    covers the specifics.
+-   `CONTRIBUTING` \
+    Directions on how to contribute code to the project, or a pointer to the Wiki page with that information.
+-   `CHANGELOG` \
+    A human readable list of recent changes. Changes should at least include the current release. This
+    file may be maintainer curated or mechanically produced.
+-   Continuous Integration / Continuous Delivery (CICD) configurations \
+    Configurations needed to run CICD on Hyperledger provided systems.
+
+## Recommended
+
+Repositories SHOULD have these files. Named files SHOULD be at the root of the repository
+
+-   `NOTICE`
+    -   As per section 4 subsection d of the
+        [Apache License, Version 2](https://www.apache.org/licenses/LICENSE-2.0)
+-   Apache License Header information in each source code file. \
+    This SHOULD include the snippet `SPDX-License-Identifier: Apache-2.0` as part of the header.
```

Comment:
[nitpick] Either the snippet `SPDX..` or text as defined in the Apache 2.0 license.

Reply:
https://wiki.hyperledger.org/display/TSC/Copyright+and+License+Policy requires the SPDX form.

Reply:
Wiki recommends `SPDX` form. Then there is a section that says

```
IMPORTANT: The license of a file, regardless of how specified, should not be modified unless agreed to by all contributors to the file.
```

There are projects which still follow the text as defined in the Apache 2.0 license. If we are mandating the format then it shall go through a round of TSC discussion.

Reply:
Adding the SPDX header to reflect the license fo the file does not change the license of a file.  And this is for "Hyperledger" code not third party code.

Reply:
Here is an example/file from Hyperledger, but not making use of SPDX header. https://github.com/hyperledger/sawtooth-core/blob/7651cb5552800dbac98559b46339a84eae472ef7/validator/src/lib.rs#L1

Example, having both SPDX and the text as defined in the Apache 2.0 license
https://github.com/hyperledger/besu/blob/ffc22522f8052645ae7453dcc394a70a0f6f797e/besu/src/main/java/org/hyperledger/besu/Besu.java#L1

Example, with only SPDX
https://github.com/hyperledger/fabric/blob/62cd59ca2ee7e37f82ed7e9bc7b08cb9f6822df0/core/peer/peer.go#L4

What I meant here is to accept any of these. Wiki too doesn't mandate one format over the other. Let's discuss pros/cons before putting this as a mandatory requirement here in the common repository structure document.

Reply:
Per TSC 2021-07-15 I added clarification it applies to new files.